### PR TITLE
GH-431 Replace BaseFederatedRepoSchema with BaseLocalRepoSchema for federated repos

### LINKED ISSUE #431: federated repositories missing properties supported in local repositories
**Describe the bug**
A Federated repository should have the same set of properties supported as a local repository resource.

**Requirements for and issue**
```shell
terraform version
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/jfrog/artifactory v6.5.1

curl -s https://artifactory.com/artifactory/api/system/version | jq .version
"7.35.2"
```

```hcl2
resource "artifactory_federated_docker_repository" "my" {
  key             = "my"
  max_unique_tags = 10
  member {
    url     = "http://localhost"
    enabled = true
  }
}

Error: Unsupported argument
│ 
│   on federated.tf line 247, in resource "artifactory_federated_docker_repository" "my":
│  247:   max_unique_tags = 10
│ 
│ An argument named "max_unique_tags" is not expected here.
```
**Expected behavior**
The only difference between local and federated resources should be federation members. All other repository properties must be the same.

**Additional context**
Here is a full list of discrepancies between local and federated generated using [tfschema](https://github.com/minamijoyo/tfschema) tool

script:
```shell
types=$(tfschema resource list artifactory | grep federated | cut -d_ -f3)
for type in $types; do
    if [[ "${type}" == "docker" ]]; then
        continue
    fi
    echo $type
    diff <(tfschema resource show -format=json artifactory_local_${type}_repository | jq --sort-keys .attributes[].name) <(tfschema resource show -format=json artifactory_federated_${type}_repository | jq --sort-keys .attributes[].name)
    echo ""
done
echo docker_v1
diff <(tfschema resource show -format=json artifactory_local_docker_v1_repository | jq --sort-keys .attributes[].name) <(tfschema resource show -format=json artifactory_federated_docker_repository | jq --sort-keys .attributes[].name)
echo docker_v2
diff <(tfschema resource show -format=json artifactory_local_docker_v2_repository | jq --sort-keys .attributes[].name) <(tfschema resource show -format=json artifactory_federated_docker_repository | jq --sort-keys .attributes[].name)
```
output:
```
alpine
8d7
< "index_compression_formats"
12d10
< "primary_keypair_ref"

bower

cargo
1d0
< "anonymous_access"
9d7
< "index_compression_formats"

chef

cocoapods

composer

conan

conda

cran

debian
8d7
< "index_compression_formats"
12d10
< "primary_keypair_ref"
18,19d15
< "secondary_keypair_ref"
< "trivial_layout"

gems

generic

gitlfs

go

gradle
3d2
< "checksum_policy_type"
7,8d5
< "handle_releases"
< "handle_snapshots"
12d8
< "max_unique_snapshots"
20,21d15
< "snapshot_version_behavior"
< "suppress_pom_consistency_checks"

helm

ivy
3d2
< "checksum_policy_type"
7,8d5
< "handle_releases"
< "handle_snapshots"
12d8
< "max_unique_snapshots"
20,21d15
< "snapshot_version_behavior"
< "suppress_pom_consistency_checks"

maven
3d2
< "checksum_policy_type"
7,8d5
< "handle_releases"
< "handle_snapshots"
12d8
< "max_unique_snapshots"
20,21d15
< "snapshot_version_behavior"
< "suppress_pom_consistency_checks"

npm

nuget
6d5
< "force_nuget_authentication"
10d8
< "max_unique_snapshots"

opkg

puppet

pypi

rpm
3d2
< "calculate_yum_metadata"
6d4
< "enable_file_lists_indexing"
13d10
< "primary_keypair_ref"
19d15
< "secondary_keypair_ref"
21,22d16
< "yum_group_file_names"
< "yum_root_depth"

sbt
3d2
< "checksum_policy_type"
7,8d5
< "handle_releases"
< "handle_snapshots"
12d8
< "max_unique_snapshots"
20,21d15
< "snapshot_version_behavior"
< "suppress_pom_consistency_checks"

vagrant

docker_v1
1d0
< "api_version"
4d2
< "block_pushing_schema1"
11d8
< "max_unique_tags"
19d15
< "tag_retention"
docker_v2
1d0
< "api_version"
4d2
< "block_pushing_schema1"
11d8
< "max_unique_tags"
19d15
< "tag_retention"
```

### PR DESCRIPTION
**Pull request by alexhung**
Fixes #431 

Update README with correct host name for testing federated repos

### CODE REVIEW

**Comment by alexhung** [comment]
@chb0github ☝🏼

**Review by chb0github — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ The [scripts/run-artifactory.sh](scripts/run-artifactory.sh) starts two Artifact
 
 Set the env var to the second Artifactory instance URL. This is the URL that will be accessible from `artifactory-1` container (not the URL from the Docker host):
 ```sh
-$ export ARTIFACTORY_URL_2=http://localhost:9081
+$ export ARTIFACTORY_URL_2=http://artifactory-2:8082
```
Comment (danielmkn) [inline README.md:121]: is this correct? 8082 will be already taken by the first instance

Comment (alexhung) [reply README.md:121]: Yes, this is the URL used by the first docker instance to access the second one. They are in the same Docker network so can be accessed via the service name and port.

File: pkg/artifactory/resource/repository/federated/resource_artifactory_federated_generic_repository.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ import (
 
 func ResourceArtifactoryFederatedGenericRepository(repoType string) *schema.Resource {
 
-	var federatedSchema = util.MergeSchema(BaseFederatedRepoSchema, map[string]*schema.Schema{
+	var federatedSchema = util.MergeSchema(local.BaseLocalRepoSchema, map[string]*schema.Schema{
```
[the remark this answers is not in the document]
Comment (alexhung) [reply pkg/artifactory/resource/repository/federated/resource_artifactory_federated_generic_repository.go:15]: Good catch. Let me check.

**Review by chb0github — APPROVED** [review]

**Review by danielmkn — APPROVED** [review]

**Pull request merged**